### ISSUE TITLE
Do not load plugins when QWebSettings::PluginsEnabled is false

### ISSUE THREAD

**Issue by trollixx** [annulen/webkit#755]
We've run into several issues ([1] and [2]) with Qt WebKit crashing due to the fact, that it tries to load plugins regardless of `QWebSettings::PluginsEnabled` is `false` (default value). I think the correct behavior should be Qt WebKit not touching the plugins in the first place.

**Comment by trollixx** [comment]
I guess a possible workaround would be unsetting `QWebSettings::pluginSearchPaths`, but I haven't tested that yet.

**Comment by annulen** [comment]
Probably `PlatformStrategiesQt::getPluginInfo` should avoid calling `PluginDatabase::plugins()` and following code when `QWebSettings::PluginsEnabled` is `false`

**Comment by annulen** [comment]
Should be trivial

**Comment by trollixx** [comment]
I guess I'll submit a patch to Gerrit. Should it also respect value of `QWebSettings::PluginsEnabled()` in `PlatformStrategiesQt::refreshPlugins()`?

**Comment by kevinhendricks** [comment]
Based on this discussion here, and based on some changes made in Qt 5.4's webkit to help address this issue, we use the following patch to disable plugin loading if PluginsEnabled is set to false.

https://github.com/Sigil-Ebook/Sigil/blob/master/docs/qtwebkit_no_load_plugins_when_disabled.patch

We include it in our own QtWebkit builds for Sigil and it appears to work.

Please consider adding this to your next release.

Hope this helps,

**Comment by annulen** [comment]
This patch is not acceptable in its current form, because setting for QWebPage may differ from global

**Comment by kmatheussen** [comment]
The patch is also 404 now.

**Comment by kevinhendricks** [comment]
Yes, we have completely moved to QtWebEngine which does not have this issue.

Here is that patch under the tag for our last release of Sigil that used QtWebKit if you want it.
https://github.com/Sigil-Ebook/Sigil/blob/0.9.14/docs/qtwebkit_no_load_plugins_when_disabled.patch
